File: samples/java/src/main/java/com/example/samples/java/MainActivity.java
package com.example.samples.java;

import android.os.Bundle;
import android.util.Log;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;

import com.example.java.Continuation;
import com.example.core.Client;
import com.example.core.response.Repo;

import java.util.List;

public class MainActivity extends AppCompatActivity {

    Client client = new Client();

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        client.getReposAsync("benjaminledet",  new Continuation<List<Repo>>() {

            @Override
            public void resume(List<Repo> value) {
                Log.d("MainActivity", value.toString());
            }

            @Override
            public void resumeWithException(@Nullable  Throwable throwable) {
                Log.e("MainActivity", "getReposAsync", throwable);
            }
        });
    }
}

File: samples/kotlin/src/main/java/com/example/samples/kotlin/MainActivity.kt
package com.example.samples.kotlin

import android.os.Bundle
import android.util.Log
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import com.example.core.Client
import kotlinx.coroutines.launch

class MainActivity : AppCompatActivity() {

    private val githubClient = Client()

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        lifecycleScope.launch {
            try {
                val result = githubClient.getReposAsync("benjaminledet")
                Log.d("MainActivity", result.toString())
            } catch (exception: Exception) {
                Log.e("MainActivity", "getReposAsync", exception)
            }
        }
    }
}
